lua/nvim-tree-preview/preview.lua

```diff
@@ -486,32 +486,56 @@ end
 ---@param size {width: number, height: number} Desired window dimensions
 ---@return {row: number, col: number} Calculated position
 function Preview:calculate_win_position(tree_win, size)
-  local view_side = require('nvim-tree').config.view.side
+  local row, col
+  local calculate_row, calculate_col = true, true
+
+  -- Check for user config first
+  if config.win_position then
+    if config.win_position.row ~= nil then
+      row = type(config.win_position.row) == 'function' and config.win_position.row(tree_win, size)
+        or config.win_position.row
+      calculate_row = false
+    end
 
-  -- Get cursor and window scroll information
-  local cursor_row = vim.api.nvim_win_get_cursor(tree_win)[1] - 1 -- Convert from 1-based to 0-based
-  local win_info = vim.fn.getwininfo(tree_win)[1]
-  local topline = win_info.topline - 1 -- Convert from 1-based to 0-based
+    if config.win_position.col ~= nil then
+      col = type(config.win_position.col) == 'function' and config.win_position.col(tree_win, size)
+        or config.win_position.col
+      calculate_col = false
+    end
+  end
 
-  -- Calculate cursor position relative to visible window
-  local relative_cursor = cursor_row - topline
+  if calculate_row or calculate_col then
```

Comment:
Rather than using the intermediate `calculate_row`/`calculate_col` variables, what if we just check if `row` or `col` is `nil` here?